modules/java-modules/hdfs/src/main/java/org/syslog_ng/hdfs/HdfsDestination.java

```diff
@@ -289,20 +289,25 @@ private void closeHdfs() {
         }
     }
 
+    private void archiveFile(HdfsFile hdfsfile) {
+        Path filePath = hdfsfile.getPath();
+        logger.debug(String.format("Trying to archive %s to %s", filePath.getName(), options.getArchiveDir()));
```

Comment:
I forgot previously, although it was my [patch](https://github.com/balabit/syslog-ng/commit/a35d5c1b09c63b0900bf1744ebf82be4f46df806):
if you use `options.getArchiveDir()` here instead of the `archiveDir` attribute, then you will override a previously disabled archiving.
I had an [alternative](https://github.com/gaborznagy/syslog-ng/commit/bb98c77b9d9944ae263c0a60c34087254570bd5b) to my patch, but I did not like it.

Reply:
Thanks for fixing.